### PR TITLE
Use multi-stage build

### PR DESCRIPTION
**Pull request by bewt85**
Hi @dcu 

I was wondering if you're interested in using multistage builds?  From Docker 17.05+ you can now build the golang binary in one container and then load it into another container automatically in a single `docker build` step without sacrificing image size.

https://docs.docker.com/engine/userguide/eng-image/multistage-build/

The advantage is that you could then auto-build the container in docker hub.

Let me know if you'd like me to make any other changes.

Thanks for building the project.

Ben

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,15 @@
+FROM       golang:alpine as builder
+
+RUN apk --no-cache add curl git make perl
+RUN curl -s https://glide.sh/get | sh
+RUN git clone https://github.com/dcu/mongodb_exporter.git /go/src/github.com/dcu/mongodb_exporter
```
Comment (dcu) [inline Dockerfile]: does this need to be an absolute path?

Comment (bewt85) [reply Dockerfile]: I have to confess that I wasn't sure so I err'd on the side of caution / went with what worked first.

Comment (bewt85) [reply Dockerfile]: Thanks for reviewing this so quickly!

**Comment by bewt85** [comment]
Thanks David

**Comment by bewt85** [comment]
Would you be interested in creating a Docker Hub automated build from this so that we can just `docker pull dcu/mongodb_exporter`?  I did it once a while ago.  I think you just sign up for an account, point it at the repo and then it automagically builds each new version for you (for free if I recall correctly).

**Pull request merged**